autoload/vsnip/session/snippet/node/variable.vim

```diff
@@ -47,7 +47,7 @@ function! s:Variable.resolve() abort
     return expand('%:p:t')
 
   elseif self.name ==# 'TM_FILENAME_BASE'
-    return substitute(expand('%:p:t'), '\..*$', '', 'g')
+    return expand('%:p:t:r')
```

Comment:
Could you add one more `:r`?
I want to support the below cases.

- index.etc.test.tsx -> index.etc
- index.test.tsx -> inde
- index.tsx -> index
- .vimrc -> .vimrc

Reply:
Did you want to remove all of the extensions of the filename?
If you so, I'll use `substitute()` with regexp, `^\S\{-}\zs\..*$`, instead of adding `:r`. This regexp can remove all the extensions but doesn't become an empty string with dotfiles.

Reply:
Supplementary explanation:
I thought you wanted to support those first:
* index.etc.test.tsx -> index
* index.three.or.more.extensions -> index

Reply:
Yes, Your suggested implementation is good for me!